pylint/checkers/imports.py

```diff
@@ -735,9 +740,15 @@ def _internal_dependencies_info(self):
 
     def _check_wildcard_imports(self, node):
         for name, _ in node.names:
-            if name == '*':
+            if name == '*' and not self._wildcard_import_is_allowed(node):
                 self.add_message('wildcard-import', args=node.modname, node=node)
 
+    def _wildcard_import_is_allowed(self, node):
+        if not self.config.allow_wildcard_with_all:
+            return False
+        module = node.do_import_module()
```

Comment:
The module could be passed instead to this method, so that we won't do the import for every name. Also, make sure to handle the errors for this import, since it might fail.